server/src/index.ts

```diff
@@ -1,40 +1,24 @@
-import express from 'express'
-import { RequestContext } from '@mikro-orm/core'
-import { MikroORM } from '@mikro-orm/core/MikroORM'
-import { v1Router } from './shared/infra/http/routes'
-import mikroORMConfig from './mikro-orm.config'
-import { DB } from './shared/infra/db'
-import { UserEntity } from './shared/infra/db/entities/user.entity'
-import { ClubEntity } from './shared/infra/db/entities/club.entity'
-import { EventEntity } from './shared/infra/db/entities/event.entity'
-import { TagEntity } from './shared/infra/db/entities/tags/tag.entity'
+import { setupMikroDB } from './setup/db'
+import { setupApplication } from './setup/application'
+import { setupWaterparkExpress } from './setup/http/express'
 
-const app = express()
-
-const port = process.env.PORT || 3001
-
-const initializeORM = async (DBObject: typeof DB) => {
-  DBObject.orm = await MikroORM.init(mikroORMConfig)
-  DBObject.em = DBObject.orm.em
-  DBObject.usersEntityRepo = DBObject.orm.em.getRepository(UserEntity)
-  DBObject.clubsEntityRepo = DBObject.orm.em.getRepository(ClubEntity)
-  DBObject.eventsEntityRepo = DBObject.orm.em.getRepository(EventEntity)
-  DBObject.tagsEntityRepo = DBObject.orm.em.getRepository(TagEntity)
-  const migrator = DBObject.orm.getMigrator()
-  await migrator.up()
+interface WaterparkOptions {
+  port: string
 }
 
-const bootstrap = async () => {
-  await initializeORM(DB)
-  app.use(express.json())
+const waterpark = async (options: WaterparkOptions) => {
+  const { orm, repos } = await setupMikroDB()
+  const migrator = orm.getMigrator()
+  await migrator.up()
 
-  app.use((_req, _res, next) => RequestContext.create(DB.orm.em, next))
-  app.use('/api/v1', v1Router)
+  const { controllers } = setupApplication(repos)
 
-  app.use((_req, res) => res.status(404).json({ message: 'No route found' }))
-  app.listen(port, () => {
+  const { app } = setupWaterparkExpress(controllers, { mikroORM: orm })
```

Comment:
The key `app` on the return type of `setupWaterparkExpress` is confusing. Since the function is always called after `setupApplication`, is isn't immediately clear what's the word "application" means in our codebase.

If you look in the definition of `setupWaterparkExpress()`, there's even a function call named `setupBasicApp`. This is confusing because I thought the "app" was setup after `setupApplication` which is from `./setup/application`.

Does renaming all instances of the word `app` in `./setup/http/express`make sense?

```suggestion
  const { expressServer } = setupWaterparkExpress(controllers, { mikroORM: orm })
```

Reply:
Great suggestion!